### ISSUE TITLE
When an enum is imported and used only as a type with "verbatimModuleSyntax" enabled, an error is not being shown

### ISSUE THREAD

**Issue by lsalazarm99** [microsoft/TypeScript#55741]
### 🔎 Search Terms

"verbatimModuleSyntax", "enum"

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about "verbatimModuleSyntax" (there wasn't any) and "enums".

### ⏯ Playground Link

https://www.typescriptlang.org/dev/bug-workbench/?verbatimModuleSyntax=true&ts=5.0.4#code/PTAEAEDcFMCcCMCGAXAlgWwLIHsAmBXAG2gGUBPAO2UQA8AuUZWfaAKFZAgDNViLF00Bj2IBaAO6pkAC1HQaAB2yxkAOmQBnVvKUrQ0CvnSgA6stwbQAb1ag7oAIKgAvKABEDtwBpb9gEIu7n7erAC+7DrKyIxkCtCm5oEaTKgUAOYA3Oyc4CIGAkKgqbjy6loYutFWCbAWXjW4oKGgXLDYxm6qwHkSUrKRKm7sxNGIDGa1WSOgiBRkE7jj5hoZQA

### 💻 Code

```ts repro
// @verbatimModuleSyntax: true

// @filename: file-with-export.ts
export enum Words {
    A = "A",
    B = "B",
}

export type Word = string;

// @filename: index.ts
import { Words, Word } from "./file-with-export"

let a: Word;
let anyWord: Words;
```

### 🙁 Actual behavior

The import of the `Words` enum doesn't show any errors.

### 🙂 Expected behavior

The import of the `Words` enum should show a 1484 error.

### Additional information about the issue

The mentioned import should show an error because it's being used only as a type and the "verbatimModuleSyntax" flag is enabled.

**Comment by fatcerberus** [comment]
It’s being *used* only as a type, but it’s a valid runtime import, so I don’t see why this would be an error.

**Comment by lsalazarm99** [comment]
🤔 I understand your point, and I think you are right. In that case, how can I get the effects of the "importsNotUsedAsValues" flag when marked as "error"? Since it's going to be deprecated in favor of "verbatimModuleSyntax", I would expect the latter to handle its functionality.

**Comment by MartinJohns** [comment]
> Since it's going to be deprecated

Where did you read that?

**Comment by lsalazarm99** [comment]
It's in the documentation of the rule, [here](https://www.typescriptlang.org/tsconfig#importsNotUsedAsValues).

**Comment by RyanCavanaugh** [comment]
There isn't a new configuration that corresponds to the old behavior of erroring here, since no "problem" per se has occurred (only a style violation). https://github.com/microsoft/TypeScript/pull/52203#issuecomment-1476559412

**Comment by fatcerberus** [comment]
> `verbatimModuleSyntax` means "only `type` annotated imports are removed" and that's it.

Hmm, shouldn’t the unannotated import of `Word` (**not** `Words`) be an error then, since that’s a pure type and thus can’t be imported at runtime?

**Comment by RyanCavanaugh** [comment]
> Hmm, shouldn’t the unannotated import of Word (not Words) be an error then

It is

![image](https://github.com/microsoft/TypeScript/assets/6685088/9edd223a-0be0-4a7a-94c9-60b04a392186)

**Comment by jakebailey** [comment]
It is:

![image](https://github.com/microsoft/TypeScript/assets/5341706/9091568e-956b-4625-9a17-67c092eb9731)

**Comment by RyanCavanaugh** [comment]
My cropping is better

**Comment by jakebailey** [comment]
I had a colon

**Comment by MartinJohns** [comment]
@jakebailey I hope you still have a colon.

**Comment by fatcerberus** [comment]
> It is

Ah okay, good.  I wasn’t sure because the OP only mentioned the “missing” error on `Words` so I just assumed the rest of the repro code was legal.  I probably should have clicked the playground link before commenting 😅

**Comment by lsalazarm99** [comment]
Got it! Thanks for your help! I think the new flag is still a bit confusing. I was expecting it to be able to do the same things that the deprecated flag does with the "error" option.

As @jakebailey said, @typescript-eslint has the situation covered. By the way, I think you were talking about the [@typescript-eslint/consistent-type-imports](https://typescript-eslint.io/rules/consistent-type-imports/) rule, since it is the one that makes the expected error be shown.

**Comment by jakebailey** [comment]
Some combo of those rules, sure. You do probably want the rule I mentioned if you want to be warned that `import { type A } from "..."` will turn into `import {} from "..."` at runtime.
